libcontainer/specconv/spec_linux.go

```diff
@@ -332,7 +332,7 @@ func CreateLibcontainerConfig(opts *CreateOpts) (*configs.Config, error) {
 	if err != nil {
 		return nil, err
 	}
-	cwd, err := filepath.Abs(rcwd)
+	cwd, err := filepath.EvalSymlinks(rcwd)
```

Comment:
🤔I'm not sure which is better, `filepath.EvalSymlinks` after `os.Getwd` can keep consistent with validation on [validator.go](https://github.com/opencontainers/runc/blob/657ed0d4a0ce3c46e202ef54e6baf0d5e88f2c01/libcontainer/configs/validate/validator.go#L52) and no need to import a new package, but seems `syscall.Getwd` is a briefer solution

Reply:
Yeah, `unix.Getcwd` should just give you the full path, no funny business.

Reply:
> no funny business.

Except it might prepend `(unreachable)` to the path, essentially making it a relative path, which might work as a path relative to cwd (glibc wrapper checks for that and returns ENOENT). _Maybe_ RHEL kernels changed this from prepending to appending, which fixes this particular issue (or perhaps I am mixing this up with `(deleted)` prefix/suffix as shown in /proc/$PID/fd/$N for removed files).

> no need to import a new package, but seems `syscall.Getwd` is a briefer solution

Package `syscall` is obsoleted in favor of `x/sys/unix`, and we gradually switch to using `x/sys/unix`, but practically in this case either one is fine.

In either case, we need to implement a wrapper like this
```go
// getwd is a wrapper similar to os.Getwd, except it always gets
// the value from the kernel, which guarantees the returned value
// to be absolute and clean.
func getwd() (dir string, err error)
       for {
                dir, err = syscall.Getwd()
                if err != syscall.EINTR { //golint:nolint // syscall errors are bare
                       break
                }
       }
       if err == nil && dir[0] != '/' {
                // In case cwd is not accessible, Linux kernel can return path
                // string that starts with "(unreachable)", making it a potentially
                // valid relative path. Fix the issue by returning an error.
                err = os.ENOENT
       }
       return dir, os.NewSyscallError("getwd", err)
}
```

Surely, this needs to be fixed in `syscall` and `x/sys/unix` as well:
* https://go-review.googlesource.com/c/go/+/387174
* https://go-review.googlesource.com/c/sys/+/387194

Reply:
@kolyshkin Are you sure this is the case even with the `getcwd(2)` syscall? I'll take a look at the kernel sources later...

Reply:
Ah, I wasn't aware of that behaviour. I just checked and yeah there are a handful of cases where you get something else prepended. Seems like it would make more sense to return a positive integer in those cases to indicate there was some non-fatal issue with the path, but I digress...